pkg/pillar/vendor/github.com/digitalocean/go-qemu/qmp/socket.go

```diff
@@ -135,13 +135,29 @@ func (mon *SocketMonitor) Connect() error {
 		return err
 	}
 
-	// Check for no error on return
-	var r response
-	if err := dec.Decode(&r); err != nil {
-		return err
-	}
-	if err := r.Err(); err != nil {
-		return err
+	// Wait for response
+	scanner := bufio.NewScanner(mon.c)
+	for scanner.Scan() {
+		var e Event
+
+		b := scanner.Bytes()
+		if err := json.Unmarshal(b, &e); err != nil {
+			return err
+		}
+
+		// If not an event then our qmp capabilities response
+		if e.Event == "" {
```

Comment:
In the case of receiving JSON data without an Event field, should we consider additional validation to ensure the data is still relevant to our needs, or do we only expect event-related JSON in this asynchronous flow?

Reply:
all possible validation is below in the same if branch

Reply:
There, we assume that we wait for any reply that is a valid JSON and does not contain Error, right? Okay, the rest of the checks are up to the caller, I guess.